app-android/features/chat/src/main/kotlin/com/closet/features/chat/ai/OpenAiChatProvider.kt

```
@@ -0,0 +1,132 @@
+
+    /**
+     * Builds the full chat-completions endpoint URL from the user-supplied base.
+     *
+     * Rules:
+     * - Blank → use [DEFAULT_BASE_URL] + `/v1/chat/completions`.
+     * - Base has no path (or only `/`) → append `/v1/chat/completions`.
+     * - Base already has a non-trivial path → append only `/chat/completions`.
+     */
+    private fun buildCompletionsUrl(raw: String): String {
+        val base = (if (raw.isBlank()) DEFAULT_BASE_URL else raw).trimEnd('/')
+        val path = try { java.net.URI(base).path ?: "" } catch (e: Exception) { "" }
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)